server/src/main/java/org/opentosca/toscana/plugins/kubernetes/visitor/imgtransform/DockerfileBuildingVisitor.java

```diff
@@ -170,7 +174,31 @@ private void handleDefault(RootNode node) {
                     logger.warn("Failed reading Port from node {}", node.getEntityName(), ex);
                 }
             });
+
+            //Temporarily set the private address to localhost (127.0.0.1) if there is a connection in the same pod
+            //The same compute node
+            node.getRequirements().forEach(e -> {
+                if (e.getRelationship().isPresent() && e.getRelationship().get() instanceof ConnectsTo) {
+                    for (Object o : e.getFulfillers()) {
+                        if (o instanceof RootNode) {
+                            NodeStack targetStack = this.connectionGraph.vertexSet()
+                                .stream().filter(ek -> ek.hasNode((RootNode) o)).findFirst().orElse(null);
+                            if (targetStack != null &&
+                                targetStack.getComputeNode() == this.stack.getComputeNode()) {
+                                adresses.put(this.stack, this.stack.getComputeNode().getPrivateAddress().get());
```

Comment:
How do you know that the Optional is present? You might want to replace the `get()` with `getOrElse(null)`.

Reply:
I Set The Private address of every compute node beforehand. (After the initial grouping in pods)